test/ProjectService.Business.UnitTests/Commands/EditTaskCommandTests.cs

```diff
@@ -0,0 +1,345 @@
+                new Operation<EditTaskRequest>(
+                    "replace",
+                    $"/{nameof(EditTaskRequest.PriorityId)}",
+                    "",
+                    _priorityId),
+                new Operation<EditTaskRequest>(
+                    "replace",
+                    $"/{nameof(EditTaskRequest.StatusId)}",
+                    "",
+                    _statusId),
+                new Operation<EditTaskRequest>(
+                    "replace",
+                    $"/{nameof(EditTaskRequest.TypeId)}",
+                    "",
+                    _typeId),
+            }, new CamelCasePropertyNamesContractResolver());
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            #region Mock initialization
+
+            _taskRepositoryMock = new Mock<ITaskRepository>();
+            _projectRepositoryMock = new Mock<IProjectRepository>();
+            _validatorMock = new Mock<IEditTaskValidator>();
+            _accessValidatorMock = new Mock<IAccessValidator>();
+            _httpAccessorMock = new Mock<IHttpContextAccessor>();
+            _mapper = new PatchDbTaskMapper();
+            _loggerMock = new Mock<ILogger<EditTaskCommand>>();
+            _requestClient = new Mock<IRequestClient<IGetDepartmentRequest>>();
+            _operationResultBrokerMock = new Mock<Response<IOperationResult<IGetDepartmentResponse>>>();
+            
+            #endregion
+
+            ClientRequestUp(Guid.NewGuid());
+            RcGetDepartment(Guid.NewGuid());
+            
+            #region Mock default setups
+
+            _taskRepositoryMock
+                .Setup(x => x.Edit(new DbTask
+                    {
+                        Id = _taskId
+                    },
+                    _mapper.Map(_request)))
+                .Returns(true).Verifiable();
+
+            _taskRepositoryMock
+                .Setup(x => x.Get(_taskId))
+                .Returns(new DbTask
+                {
+                    ProjectId = _projectId
+                }).Verifiable();
+            
+            _accessValidatorMock
+                .Setup(x => x.IsAdmin())
+                .Returns(false);
+            
+            _projectRepositoryMock.Setup(x => x.GetProject(_projectId)).Returns(new DbProject()
+            {
+                Users = new List<DbProjectUser>()
+                {
+                    new DbProjectUser
+                    {
+                        UserId = Guid.NewGuid()
+                    }
+                }
+            }).Verifiable();
+
+            #endregion
+            
+            _command = new EditTaskCommand(
+                _taskRepositoryMock.Object,
+                _projectRepositoryMock.Object,
+                _validatorMock.Object,
+                _accessValidatorMock.Object,
+                _httpAccessorMock.Object,
+                _mapper,
+                _loggerMock.Object,
+                _requestClient.Object);
+        }
```

Comment:
use autoMocker

Reply:
As I know, we can use just mock or automock as author want to

Reply:
Egor is agree